packages/vite/src/node/plugins/css.ts

```diff
@@ -1504,8 +1504,10 @@ function loadPreprocessor(
     return (loadedPreprocessors[lang] = _require(resolved))
   } catch (e) {
     if (e.code === 'MODULE_NOT_FOUND') {
+      const packageManager =
+        process.env.npm_config_user_agent?.split(' ')[0].split('/')[0] || 'npm'
```

Comment:
I hope that the `[pkgFromUserAgent]`(https://github.com/vitejs/vite/blob/13ac37df8e135fe6b4990fac6f149a36a19e5409/packages/create-vite/src/index.ts#L450) function can be reused in the same way as `import { pkgFromUserAgent } from '@vite/create-vite'` or `import { pkgFromUserAgent } from '@vite/shared'`, but currently, it is not exposed in that way.

Reply:
I think it's ok to duplicate it for now. It would be nice to move this into `utils.ts` though, and have a separate function that returns package manager specific commands.

e.g. with `yarn` and `pnpm`, we want `yarn add` and `pnpm add` instead. That way we can do something like `${installCommand} -D ${lang}` instead.

Reply:
Oh right... I completely forgot that `yarn` does not support the `i`/`install` command. However, `pnpm` does support `pnpm i <package_name>` as an alias for `pnpm add <package_name>`. Nonetheless, `pnpm add` is the more officially supported method.

I will refactor it to `utils.ts` and make it more verbose. Thank you for your suggestion!